### DISCUSSION TITLE
Groupable/stackable windows

### DISCUSSION THREAD

**Discussion by ghost** [GreenteaOS/Tofita#18] [Ideas]
I share some suggestions that I believe could add value and novelty to the operating system. Below, I will describe two proposed functionalities:

Window Tiles: This feature would allow dividing a window into multiple tiles, each displaying a different application without overlapping. It would be like having multiple windows open at the same time but within a single window. The main goal would be to improve workspace organization and make the most of the screen, especially on devices with large displays. Users could easily view and access each tile. It's worth mentioning that this functionality goes beyond the "Aero Snap" feature in Windows, which only allows dragging windows and placing them in the corners of the screen without unifying them into a single window. An additional advantage of Window Tiles is that the window would maintain a proportional size relationship when resizing a tile, automatically adjusting the size of the others. Additionally, each window added to a Window Tile would have an icon in the top right corner that would allow separating it from the tile and returning it to its normal window state, if desired by the user.

Stacking Windows: This functionality would allow having multiple windows within a single window, with the ability to switch between them similar to tabs in a web browser. It would be like having multiple tabs open simultaneously but within a single window. The main goal would be to simplify window management and save screen space. Users could create, close, and rearrange the tabs according to their preferences.

Both functionalities could be used independently or combined, depending on the user's preferences and needs. For example, a user could have a main window with several window tiles, each displaying a different application, and within each window tile, they could have multiple stacking windows, each showing a different tab. This would allow quick access to multiple applications and tabs with just a few clicks, without losing the overall view of the workspace, thus creating a more personalized and flexible environment.

**Comment by PeyTy** [comment]
Hello! Damn I should add discussion notifications to the bot. I totally missed your post.

@SrWesly is this the same idea behind Haiku OS? Or maybe you were inspired by some other project? I think I'll create a prototype and we'll see how it goes. The concept is interesting, but it may provide less benefits in practice. But it works for browsers, so it may work for us too!

**Reply by ghost** [reply]
Yeah, I was inspired by Haiku OS. Maybe in Greentea something similar could be added. I also suggest "sticky" windows whose function would let you make regular windows and stacked windows stick together and share resizing. This on large screens would work well. I represented the ideas in MS Paint. It can be done better, I just did it fast
![Stackable windows](https://github.com/GreenteaOS/Tofita/assets/85891667/15bd4f1c-8b1e-4238-9cd3-bf8cf212eab2)
![Sticky windows concept for Greentea OS](https://github.com/GreenteaOS/Tofita/assets/85891667/d4ad1337-4473-443d-a524-2848f1a8252f)

**Reply by PeyTy** [reply]
Oh, never thought about having tiles inside a window. Cool

**Reply by ghost** [reply]
That is a better term. I will update my suggestion accordingly
